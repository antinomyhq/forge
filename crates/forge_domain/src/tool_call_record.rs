use derive_setters::Setters;
use serde::{Deserialize, Serialize};

use crate::{ToolCallFull, ToolResult};

/// Represents a complete tool invocation cycle, containing both the original
/// call and its corresponding result.
#[derive(Clone, Debug, Deserialize, Serialize, Setters)]
#[setters(strip_option, into)]
pub struct ToolCallRecord {
    pub tool_call: ToolCallFull,
    pub tool_result: ToolResult,
}

impl ToolCallRecord {
    /// Creates a new CallRecord from a tool call and its result
    pub fn new(call: ToolCallFull, result: ToolResult) -> Self {
        Self { tool_call: call, tool_result: result }
    }
}
