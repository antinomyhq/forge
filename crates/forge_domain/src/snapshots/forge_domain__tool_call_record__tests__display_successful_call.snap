---
source: crates/forge_domain/src/tool_call_record.rs
expression: record.to_string()
snapshot_kind: text
---
<tool_result tool="fs_read"><arguments><path>"/example/file.txt"</path></arguments><result><content><![CDATA[Contents of the file]]></content></result></tool_result>
